connectors/agentic-ai/src/main/java/io/camunda/connector/agenticai/aiagent/framework/langchain4j/ChatModelFactoryImpl.java

```diff
@@ -126,42 +122,47 @@ protected BedrockChatModel.Builder createBedrockChatModelBuilder(
       BedrockProviderConfiguration configuration) {
     final var connection = configuration.bedrock();
 
-    final var bedrockClientBuilder =
-        BedrockRuntimeClient.builder()
-            .credentialsProvider(
-                switch (connection.authentication()) {
-                  case AwsDefaultCredentialsChainAuthentication ignored ->
-                      DefaultCredentialsProvider.create();
-                  case AwsStaticCredentialsAuthentication sca ->
-                      StaticCredentialsProvider.create(
-                          AwsBasicCredentials.create(sca.accessKey(), sca.secretKey()));
-                })
-            .region(Region.of(connection.region()));
-
-    if (connection.endpoint() != null) {
-      bedrockClientBuilder.endpointOverride(URI.create(connection.endpoint()));
-    }
-
     final var builder =
         BedrockChatModel.builder()
-            .client(bedrockClientBuilder.build())
+            .client(createBedrockClient(connection))
             .modelId(connection.model().model());
 
-    final var modelParameters = connection.model().parameters();
-    if (modelParameters != null) {
-      final var requestParametersBuilder = BedrockChatRequestParameters.builder();
-      Optional.ofNullable(modelParameters.maxTokens())
-          .ifPresent(requestParametersBuilder::maxOutputTokens);
-      Optional.ofNullable(modelParameters.temperature())
-          .ifPresent(requestParametersBuilder::temperature);
-      Optional.ofNullable(modelParameters.topP()).ifPresent(requestParametersBuilder::topP);
+    return applyModelParametersIfPresent(connection, builder);
+  }
 
-      builder.defaultRequestParameters(requestParametersBuilder.build());
+  private BedrockChatModel.Builder applyModelParametersIfPresent(
```

Comment:
As we handle multiple chat models in this class, could we include bedrock in the method name here to have a distinction (e.g.  `applyBedrock...`)?